pkg/watermark/fetch/edge_fetcher.go

```diff
@@ -223,13 +225,13 @@ func (e *edgeFetcher) Close() error {
 	return nil
 }
 
-// getMinFromLastProcessed returns the smallest watermark among all the last processed watermarks.
-func (e *edgeFetcher) getMinFromLastProcessed(watermark int64) int64 {
-	minWm := watermark
+// GetWatermark returns the smallest watermark among all the last processed watermarks.
+func (e *EdgeFetcher) GetWatermark() wmb.Watermark {
+	minWm := int64(0)
```

Comment:
So, line 232 checks for the case of `minWm == int64(0)`. I suppose if `e.lastProcessedWm` were empty then this would get set to 0. Is that the concern? or are you just thinking it's more elegant not to have to do the `minWm == int64(0)` check?

Reply:
I prefer to assign minWM to MaxInt as the initial value instead of doing a zero check later

Reply:
I agree